lib/ui/painting.dart

```diff
@@ -3746,6 +3746,96 @@ class ImageShader extends Shader {
   void _initWithImage(_Image image, int tmx, int tmy, int filterQualityIndex, Float64List matrix4) native 'ImageShader_initWithImage';
 }
 
+/// A shader (as used by [Paint.shader]) that runs provided SPIR-V code.
+///
+/// NOTE: This API is in beta and does not yet work on web.
+/// See https://github.com/flutter/flutter/projects/207 for roadmap.
+///
+/// When initializing or updating the [floatUniforms], the length of float
+/// uniforms must match the total number of floats defined as uniforms in
+/// the shader. They will be updated in the order that they are defined.
+///
+/// For example, if there are 3 uniforms: 1 of type float, 1 type float2/vec2,
+/// and 1 of type vec3/float3, and 1 mat2x2 then the length of [floatUniforms]
+/// must be 10.
+///
+/// The uniforms could be updated as follows:
+///
+///   Example fragment shader uniforms.
+///
+///   `uniform float a;`
+///   `uniform vec2 b;`
+///   `uniform vec3 c;`
+///   `uniform mat2x2 d;`
+///
+///   Dart code to update uniforms.
+///
+///   `shader.update(floatUniforms: Float32List.fromList([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]));`
+///
+///   Results of shader uniforms.
+///
+///   a: 1
+///   b: [2, 3]
+///   c: [4, 5, 6]
+///   d: [7, 8, 9, 10] // 2x2 matrix in column-major order
+///
+class FragmentShader extends Shader {
+  @pragma('vm:entry-point')
```

Comment:
@pragma is probably best immediately before the constructor rather than before the docs for the constructor

Reply:
(also what Effective Dart defines and what IDE features work best with)

Reply:
ah thanks, moved.